main/rust/template.py

```diff
@@ -288,7 +288,13 @@ def do_build(self):
     benv["CARGO_HOME"] = str(self.chroot_cwd / ".cargo")
     # we don't want the default cross sysroot here
     with self.profile("target:native"):
-        benv["RUSTFLAGS"] = self.get_rustflags(shell=True)
+        benv["RUSTFLAGS"] = self.get_rustflags(
+            shell=True,
+            # remap debug source paths to what will become rust-src
+            extra_flags=[
+                "--remap-path-prefix=./library=/usr/lib/rustlib/src/rust/library"
+            ],
```

Comment:
i don't think this in itself will work (but i mean, test it lol) because the reason they say ./library in the first place is because of the previous remap. try /only/ removing this entirely first and seeing what it outputs into the std .rlibs and how that changes rebuilding something that links them so we can compare